lib/core/alice_chopper_response_interceptor.dart

```diff
@@ -1,97 +1,102 @@
+}
+
+class AliceChopperInterceptor implements Interceptor {
   /// AliceCore instance
   final AliceCore aliceCore;
 
   /// Creates instance of chopper interceptor
   AliceChopperInterceptor(this.aliceCore);
 
   /// Creates hashcode based on request
-  int getRequestHashCode(BaseRequest baseRequest) {
-    var hashCodeSum = 0;
-    hashCodeSum += baseRequest.url.hashCode;
-    hashCodeSum += baseRequest.method.hashCode;
-    if (baseRequest.headers.isNotEmpty) {
-      baseRequest.headers.forEach((key, value) {
-        hashCodeSum += key.hashCode;
-        hashCodeSum += value.hashCode;
-      });
-    }
-    if (baseRequest.contentLength != null) {
-      hashCodeSum += baseRequest.contentLength.hashCode;
-    }
+  int getRequestHashCode(http.BaseRequest baseRequest) {
+    final int hashCodeSum = baseRequest.url.hashCode +
+        baseRequest.method.hashCode +
+        (baseRequest.headers.entries
+            .map((MapEntry<String, String> header) =>
+                header.key.hashCode + header.value.hashCode)
+            .reduce((int value, int hashCode) => value + hashCode)) +
+        (baseRequest.contentLength?.hashCode ?? 0);
 
     return hashCodeSum.hashCode;
   }
 
   /// Handles chopper request and creates alice http call
   @override
-  FutureOr<chopper.Request> onRequest(chopper.Request request) async {
+  FutureOr<Response<BodyType>> intercept<BodyType>(
+    Chain<BodyType> chain,
+  ) async {
     try {
-      final headers = request.headers;
+      final Map<String, String> headers = chain.request.headers;
       headers['alice_token'] = DateTime.now().millisecondsSinceEpoch.toString();
-      final changedRequest = request.copyWith(headers: headers);
-      final baseRequest = await changedRequest.toBaseRequest();
+      final Request changedRequest = chain.request.copyWith(headers: headers);
+      final http.BaseRequest baseRequest = await changedRequest.toBaseRequest();
 
-      final call = AliceHttpCall(getRequestHashCode(baseRequest));
-      var endpoint = '';
-      var server = '';
+      final AliceHttpCall call = AliceHttpCall(getRequestHashCode(baseRequest));
+      final StringBuffer endpoint = StringBuffer();
+      final StringBuffer server = StringBuffer();
 
-      final split = request.url.toString().split('/');
+      final List<String> split = chain.request.url.toString().split('/');
       if (split.length > 2) {
-        server = split[1] + split[2];
+        server.writeAll([
+          split[1],
+          split[2],
+        ]);
       }
       if (split.length > 4) {
-        endpoint = '/';
-        for (var splitIndex = 3; splitIndex < split.length; splitIndex++) {
-          // ignore: use_string_buffers
-          endpoint += '${split[splitIndex]}/';
+        endpoint.write('/');
+        for (int splitIndex = 3; splitIndex < split.length; splitIndex++) {
+          endpoint.writeAll([
+            split[splitIndex],
+            '/',
+          ]);
         }
```

Comment:
Also here I thing its also possible to use `Uri.pathSegments` for constructing a endpoint